Automattic/jetpack
# RNA Connection: refactoring and simplification

#### Changes proposed in this Pull Request:
- add REST endpoint to get the authorization URL
- get rid of the `authorizationUrl` parameter, retrieve it from the REST API
- extract the user connecting functionality into the separate `ConnectUser` component

#### Jetpack product discussion
p1HpG7-bEn-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
0. Read the [README.md](https://github.com/Automattic/jetpack/pull/19837/files#diff-887d7021f0df39dd467af1107842df6c0340ebc25120890a2e5566b66060f64b) changes to see if the instructions are clear and correct, and there are no typos.
1. Disconnect Jetpack.
2. Go to "Tools -> Connection Manager" and click "Connect" to register the site. After a little wait, you'll see the In-Place iframe.
3. Confirm that the line "Site Registered" appears above the "Connect" button.
3. Do not approve the user connection, refresh the page.
4. Click "Connect" again to complete the user connection. The In-Place iframe should appear, click "Approve".
5. The "Connect" button should disappear, the page should inform you that you're fully connected.

**Please Note:** if you are testing this PR in Jetpack Monorepo, Connection Manager UI will get enabled automatically.
Otherwise, you need to create a plugin with the following code to enable it: https://gist.github.com/sergeymitr/bc005e1b893e778c03e4f43debb91c30
